### PR TITLE
Streamline IAP code with documentation updates

### PR DESCRIPTION
**Pull request by jopmiddelkamp**
Upped the IAP step folders as I've added a new section to the codelab documentation.

Before:
Screenshot 2021-12-14 at 13 52 38

After:
Screenshot 2021-12-14 at 13 51 30

## Related issue:
https://github.com/flutter/flutter/issues/90991

## Pre-launch Checklist:
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by domesticmouse** [comment]
You'll want to pull from `master` as this PR is significantly behind. Also, have you re-created the Flutter Runners for v2.8.0? I [started down that path](https://github.com/flutter/codelabs/pull/238), but it isn't yet in a sane state.

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
@jopmiddelkamp I've staged the updated codelab. Let me know when you are ready for me to land this PR and the matching codelab update.

**Comment by jopmiddelkamp** [comment]
@domesticmouse This morning I've also updated the analysis options of all steps so it requires another review. Besides that, I've also updated the code lab document to match the changes. So if you don't find anything strange in my last commit we are ready to deploy the latest version of the codelab document and this PR.

File: in_app_purchases/complete/app/analysis_options.yaml [hunk 1]
```diff
@@ -14,7 +14,9 @@ linter:
     directives_ordering: true
```
Comment (domesticmouse) [inline in_app_purchases/complete/app/analysis_options.yaml:14]: Please replace this file with a link to https://github.com/flutter/codelabs/blob/master/analysis_options.yaml

See https://github.com/flutter/codelabs/blob/master/webview_flutter/step_03/analysis_options.yaml for example usage

File: in_app_purchases/step_00/app/analysis_options.yaml [hunk 2]
```diff
@@ -14,7 +14,9 @@ linter:
     directives_ordering: true
```
Comment (domesticmouse) [inline in_app_purchases/step_00/app/analysis_options.yaml:14]: Please follow https://github.com/flutter/codelabs/blob/master/webview_flutter/step_03/analysis_options.yaml

File: in_app_purchases/step_07/app/analysis_options.yaml [hunk 3]
```diff
@@ -14,7 +14,9 @@ linter:
     directives_ordering: true
```
Comment (domesticmouse) [inline in_app_purchases/step_07/app/analysis_options.yaml:14]: Ditto

File: in_app_purchases/step_08/app/analysis_options.yaml [hunk 4]
```diff
@@ -14,7 +14,9 @@ linter:
     directives_ordering: true
```
Comment (domesticmouse) [inline in_app_purchases/step_08/app/analysis_options.yaml:14]: ditto

File: in_app_purchases/step_09/app/analysis_options.yaml [hunk 5]
```diff
@@ -14,7 +14,9 @@ linter:
     directives_ordering: true
```
Comment (domesticmouse) [inline in_app_purchases/step_09/app/analysis_options.yaml:14]: ditto

File: in_app_purchases/step_10/app/analysis_options.yaml [hunk 6]
```diff
@@ -14,7 +14,9 @@ linter:
     directives_ordering: true
```
Comment (domesticmouse) [inline in_app_purchases/step_10/app/analysis_options.yaml:14]: ditto

**Review by domesticmouse — CHANGES_REQUESTED** [review]
I'm trying to centralise analysis configuration. Please update all `analysis_options.yaml` files to point at https://github.com/flutter/codelabs/blob/master/analysis_options.yaml

**Review by domesticmouse — APPROVED** [review]
I'll land this at the same time as the updated codelab text

**Comment by domesticmouse** [comment]
One blocker, one nit:

  - In the codelab text, there is a non-english screenshot. Please replace with an english language screenshot.
  - The `firebase_core` dependency is 1.0.3 while the latest published is 1.10.6. Update dependencies?

**Comment by jopmiddelkamp** [comment]
> One blocker, one nit:
> 
> * In the codelab text, there is a non-english screenshot. Please replace with an english language screenshot.
> * The `firebase_core` dependency is 1.0.3 while the latest published is 1.10.6. Update dependencies?

Fixed both issues.

**Pull request merged**